contracts/core/Allo.sol

```diff
@@ -1,79 +1,460 @@
+            IDistributionStrategy(_distributionStrategy),
+            _initDistributionData,
+            _token,
+            _amount,
+            _metadata,
+            _owners
+        );
     }
 
-    // @todo insert clonable strategy library, including validation that an existing pool has safe strategy
+    /// @notice Creates a new pool
+    /// @param _identityId The identityId of the pool creator in the registry
+    /// @param _allocationStrategy The address of the allocation strategy
+    /// @param _distributionStrategy The address of the distribution strategy
+    /// @param _token The address of the token that the pool is denominated in
+    /// @param _amount The amount of the token to be deposited into the pool
+    /// @param _metadata The metadata of the pool
+    /// @param _owners The owners of the pool
+    function _createPool(
+        bytes32 _identityId,
+        IAllocationStrategy _allocationStrategy,
+        bytes memory _initAllocationData,
+        IDistributionStrategy _distributionStrategy,
+        bytes memory _initDistributionData,
+        address _token,
+        uint256 _amount,
+        Metadata memory _metadata,
+        address[] memory _owners
+    ) internal returns (uint256 poolId) {
+        if (!registry.isOwnerOrMemberOfIdentity(_identityId, msg.sender)) {
+            revert UNAUTHORIZED();
+        }
+
+        Pool memory pool = Pool({
+            identityId: _identityId,
+            allocationStrategy: _allocationStrategy,
+            distributionStrategy: _distributionStrategy,
+            metadata: _metadata
+        });
+
+        poolId = ++_poolIndex;
+        pools[poolId] = pool;
+
+        // initialize strategies
+        // @dev Initialization is expect to revert when invoked more than once
+        _allocationStrategy.initialize(_identityId, poolId, address(this), _initAllocationData);
+        _distributionStrategy.initialize(_identityId, poolId, address(this), _initDistributionData);
+
+        // access control
+        bytes32 POOL_OWNER_ROLE = keccak256(abi.encodePacked(poolId));
+        bytes32 POOL_ADMIN_ROLE = keccak256(abi.encodePacked(poolId, "admin"));
 
-    // creates pool locally, transfers pool amount to distribution strategy => returns poolId
-    // takes fee from user
-    // validates that the owner is actually allowed to use the identity
-    function createPool(PoolData memory /*_poolData*/ ) external pure returns (uint256) {
-        uint32 _poolId = 0;
+        // grant pool owners roles
+        for (uint256 i = 0; i < _owners.length;) {
+            _grantRole(POOL_OWNER_ROLE, _owners[i]);
+            unchecked {
+                i++;
+            }
+        }
 
-        // todo: return the poolId? what do we want to return here?
-        return _poolId;
+        // grant admin roles to pool creator
+        _grantRole(POOL_ADMIN_ROLE, msg.sender);
+        // set admin role for POOL_OWNER_ROLE
+        _setRoleAdmin(POOL_OWNER_ROLE, POOL_ADMIN_ROLE);
```

Comment:
Personally I think I'd name these opposite? It doesn't really matter but I think of admins as all the people who can control the pool, and the owner is the one who actually owns it and has ultimate say.

Reply:
You thinking something like this:

```javascript
// grant pool admin roles
for (uint256 i = 0; i < _admins.length;) {
    _grantRole(POOL_ADMIN_ROLE, _admins[i]);
    unchecked {
        i++;
    }
}

// grant owner role to pool creator
_grantRole(POOL_OWNER_ROLE, msg.sender);

// Note: do we want this? set admin role for pool creator
// _grantRole(POOL_ADMIN_ROLE, msg.sender);
```

Reply:
> Owners and administrators both have all permissions, including posting listings, editing the organization's profile, and managing the permissions of other administrators, but an owner has control over other owners as well as administrators.

LOL derp you're right @zobront  :P Will change that 
Admin are the folks who control the pool while owners can control who is an admin / not

Reply:
Changes via 487ab9d0f9dc334037401c498461016e67737814